sql/src/test/java/io/crate/analyze/validator/ValidatorTest.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Licensed to CRATE Technology GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.analyze.validator;
+
+import com.carrotsearch.ant.tasks.junit4.dependencies.com.google.common.collect.ImmutableList;
+import io.crate.metadata.*;
+import io.crate.metadata.table.ColumnPolicy;
+import io.crate.metadata.table.TableInfo;
+import io.crate.metadata.table.TestingTableInfo;
+import io.crate.planner.RowGranularity;
+import io.crate.planner.symbol.Reference;
+import io.crate.planner.symbol.Symbol;
+import io.crate.planner.symbol.Value;
+import io.crate.types.*;
+import org.apache.lucene.util.BytesRef;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class ValidatorTest {
+
+    private static ReferenceInfos referenceInfos;
+
+    private final TableIdent testTableIdent = new TableIdent(null, "test");
+    private final ReferenceIdent testReferenceIdent = new ReferenceIdent(testTableIdent, "person");
+
+    @Before
+    @SuppressWarnings("unchecked")
+    public void mockReferenceInfos() {
+        referenceInfos = mock(ReferenceInfos.class);
+        TableInfo testTableInfo = TestingTableInfo.builder(testTableIdent, RowGranularity.DOC, new Routing())
+                .add("person", DataTypes.OBJECT, null, ColumnPolicy.DYNAMIC)
+                .add("person", DataTypes.INTEGER, ImmutableList.of("age"))
+                .add("person", DataTypes.OBJECT, ImmutableList.of("name"))
+                .add("person", DataTypes.STRING, ImmutableList.of("name", "first_name"))
```

Comment:
as mentioned, we always use BytesRef instead of String, so expected values must be of type BytesRef